gapic/schema/wrappers.py

```diff
@@ -1067,6 +1078,68 @@ def try_parse_routing_rule(cls, routing_rule: routing_pb2.RoutingRule) -> Option
         params = [RoutingParameter(x.field, x.path_template) for x in params]
         return cls(params)
 
+    @classmethod
+    def resolve(cls, routing_rule: routing_pb2.RoutingRule, request: Union[dict, str]) -> dict:
+        """Resolves the routing header which should be sent along with the request.
+        This function performs dynamic header resolution, identical to what's in `_client_macros.j2`.
```

Comment:
nit: I would add a TODO referencing #2160 about refactoring the common code and then removing this sentence.

Reply:
Fixed in https://github.com/googleapis/gapic-generator-python/pull/2131/commits/4bbfd9f5b3918b9d2da1c43f4c53f1fccadf7995